Added new folder docs with config-options description on options. (#370)
hound-search/hound: docs/config-options.md

- [ConfigOptions](#configoptions)
  * [Git options](#git-options)
  * [SVN options](#svn-options)
  * [URL options](#url-options)

> ### ConfigOptions

Reviewer: Nit: 
```suggestion
# Config Options
```

> Various options available in `config.json` are used by hound.Comprehensive list of available options are provided below with some sane defaults.

Reviewer: Nit:
```suggestion
Most of Hound's behavior is defined by a single `config.json` configuration file. A list of its available options are provided below, along with their defaults.
```

keys used in the config json file are the options,description provides details about keys.Default values gives idea about value which can be used for the option.

ConfigOption | Description | Default Values
:------ | :----- | :-----
max-concurrent-indexers | defines the total number of indexers required to be used for indexing code | 2
health-check-uri |  health check url for hound | `/healthz`
dbpath | absolute file path where the `config.json` file exists| `data`
title | Title used for the application | Hound
url-pattern | composed of base url and anchor values in form of key value pairs | n/a
vcs-config | holds the version control config, default VCS used in Hound is git.Other options for VCS are svn,mercurial,bitbucket,hg, etc.Refer to `config-example.json` to get the list of vcs and usage. Below tables provide detailed options list of each type of vcs | git
repos | holds the list of repos which are required to be indexed by Hound . Each Repo is added with reponame as a Json Key with options associated with repo as values similar to example provided in `config-example.json` | n/a

> #### Git options

Reviewer: ```suggestion
## Git Options
```

List of options associated with git vcs in repos

GitOptions  | Description | Default Values
:------ | :----- | :-----
ms-between-polls | time interval to poll the repo url | 30s
detect-ref    | used to determine branch |  master branch 
ref | used to provide reference for the branch for repo| n/a

> #### SVN options

Reviewer: ```suggestion
## SVN Options
```

List of options available for SVN vcs in repos

SvnOptions  | Descriptions| Default Values
:------ | :-----| :-----
username  | user name for the svn repo | n/a
password | password to authenticate use for svn repo | n/a

> #### URL options 

Reviewer: ```suggestion
## URL Options 
```
